share/eds/retriever.go

```diff
@@ -30,6 +30,10 @@ var (
 	tracer = otel.Tracer("share/eds")
 )
 
+type RetrieverOption func(*Retriever)
+
+type CleanupFunc func(context.Context, cid.Cid) error
+
```

Comment:
Could you remove it? Because of the atomicity argument, it's not practical to do cleanup in Retriever. Cleaning should happen _only_ after the EDS gets indexed and stored on disk.